### ISSUE TITLE
Preserve body parameter name via the vendor extension

### ISSUE THREAD

**Issue by wing328** [swagger-api/swagger-parser#966]
Consider the following OAS v2 spec, the body parameter name "body" is completely lost after parsing the spec:
```
      parameters:
        - in: body
          name: body
          description: Pet object that needs to be added to the store
          required: true
          schema:
            $ref: '#/definitions/Pet'
```

To preserve the body parameter name, we can store it in the vendor extension `x-codegen-request-body-name`

**Comment by blast-hardcheese** [comment]
Just ran into this myself. Structurally, it's immaterial, as OpenAPI 3.x+ doesn't have names for body parameters. That being said, it does not offer a pleasant experience for users migrating to the new structure using the swagger converter module.

**Comment by gracekarina** [comment]
closed by #967
